_posts/2014-11-13-cockpit-plugin-tutorial.md

```diff
@@ -1,29 +1,31 @@
 ---
 title: Creating Plugins for the Cockpit User Interface
-date: '2014-11-13'
+date: '2020-05-04'
 category: tutorial
 tags: cockpit linux
 slug: creating-plugins-for-the-cockpit-user-interface
 ---
 
-*Note: This post has been updated for changes in Cockpit 0.90 and later.*
+*Note: This post has been updated on May 2020 to adjust to current Cockpit UI and development best practices.*
 
-[Cockpit is a user interface for servers](https://cockpit-project.org). And you can add stuff to that user interface. Cockpit is internally built of various components. Each component is HTML, with Javascript logic that makes it work, and CSS to make it pretty.
+[Cockpit is a user interface for servers](https://cockpit-project.org). You can easily add your own pages to that user interface, i. e. components that appear in the menu:
 
-It's real easy to create these components. Tools are components that show up in the *Tools* menu in Cockpit:
+![Navigation menu](/images/pages-menu-top.png)
 
-![Tools menu](/images/cockpit-tools-default.png)
+Lets make ourselves another one. For this tutorial you need to [install your distribution's cockpit packages](../running.html) or [build it from git](https://github.com/cockpit-project/cockpit/blob/master/HACKING.md).
```

Comment:
```suggestion
For this tutorial you need to [install your distribution's cockpit packages](../running.html) or [build it from git](https://github.com/cockpit-project/cockpit/blob/master/HACKING.md).
```

I don't understand what "Lets make ourselves another one." is supposed to mean. (It's supposed to be "Let's make another one (_of something_) ourselves." if used.)

Perhaps: "Let's add our own page now." Or just drop the sentence entirely, as it's already implied by "You can easily add your own custom pages to the navigation." directly above.

Reply:
Agreed, that's the point of the entire blog post, after all. :-)